### PR TITLE
if there are 2 signatures from the same second, then last-writer wins

### PR DESCRIPTION
**Pull request by maxtaco**
- this solves yield's problem

### CODE REVIEW

File: openpgp/keys.go [hunk 1]
```diff
@@ -421,7 +421,7 @@ EachPacket:
 			// Section 5.2.3.3, if there are several self-signatures,
 			// we should take the newer one.
 			if current != nil &&
-				(current.SelfSignature == nil || pkt.CreationTime.After(current.SelfSignature.CreationTime)) &&
+				(current.SelfSignature == nil || !pkt.CreationTime.Before(current.SelfSignature.CreationTime)) &&
```
Comment (oconnor663) [inline openpgp/keys.go]: Could we add a comment here mentioning that this otherwise-innocuous-looking check is actually really subtle?

Comment (maxtaco) [inline openpgp/keys.go]: ok done!

Comment (maxtaco) [inline openpgp/keys.go]: btw, note the profusion of comments in this file.  All subtle crazy corner conditions!

File: openpgp/keys.go [hunk 2]
```diff
@@ -419,9 +419,17 @@ EachPacket:
...
+			// we should take the newer one.  If they were both created
+			// at the same time, but one of them has keyflags specified and the
+			// other doesn't, keep the one with the keyflags. We have actually
+			// seen this in the wild (see the 'Yield' test in read_test.go).
+			// If there is a tie, and both have the same value for FlagsValid,
+			// then "last writer wins."
 			if current != nil &&
-				(current.SelfSignature == nil || pkt.CreationTime.After(current.SelfSignature.CreationTime)) &&
+				(current.SelfSignature == nil ||
+					!pkt.CreationTime.Before(current.SelfSignature.CreationTime) ||
```
Comment (oconnor663) [inline openpgp/keys.go]: Should this line go back to being `After` instead of `!Before`? I think you're accidentally handling the `Equal` case twice.

**Comment by oconnor663** [comment]
LGTM except for that question above.

File: openpgp/keys.go (hunk 2 above)
Comment (maxtaco) [inline openpgp/keys.go]: makes sense, thanks!

**Pull request merged**